ls1intum/Artemis
# `General`: Include quiz exercise information and submissions in course and exam archive

### Checklist
#### General


#### Server

### Motivation and Context


Quiz exercise information is not included in the course or exam archive making the export incomplete.

### Description
Introduced a new `QuizExerciseWithSubmissionsExportService` that exports the information. For exporting the quiz submissions I reused the implementation from the data export.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with an end date in the past and a quiz exercise
- 1 Exam with an end date in the past and a quiz exercise
- At least one submission for each quiz exercise

1. Log in to Artemis
2. Navigate to Course Administration
3. Archive the course and the exam
4. Download the archive in both cases and verify that it contains a directory for the quiz exercise.
5. This directory should contain a JSON file with information about the exercise with all questions with answers and explantions
6. For every student participation a directory should exist that contains a pdf for every answered Drag and Drop question and a txt file with the submitted answers for all MC and SC questions and a txt file with the submitted answers for all Short Answer questions.

#### Exam Mode Testing
Covered with the steps above.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
### Test Coverage






#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [StudentParticipationRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5479/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/StudentParticipationRepository.html) | 95% | ✅ |
| [CourseExamExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5479/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/CourseExamExportService.html) | 91% | ✅ |
| [DataExportQuizExerciseCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5479/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/DataExportQuizExerciseCreationService.html) | 99% | ✅ |
| [DataExportUtil.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5479/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/DataExportUtil.html) | 100% | ✅ |
| [QuizExerciseWithSubmissionsExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5479/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/QuizExerciseWithSubmissionsExportService.html) | 100% | ✅ |
